nioChecksumTest/src/main/scala/cromwell/niochecksumtest/NioChecksumTest.scala

```diff
@@ -0,0 +1,123 @@
+package cromwell.niochecksumtest
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.stream.ActorMaterializer
+import cloud.nio.impl.drs.{DrsCloudNioFileSystemProvider, GoogleDrsCredentials}
+import com.typesafe.config.ConfigFactory
+import cromwell.cloudsupport.gcp.auth.UserMode
+import cromwell.core.WorkflowOptions
+import cromwell.core.io.{AsyncIo, IoCommandBuilder, IoPromiseProxyActor}
+import cromwell.engine.io.IoActor
+import cromwell.engine.io.IoActor.IoConfig
+import cromwell.filesystems.drs.{DrsPath, DrsReader}
+import cromwell.services.ServiceRegistryActor
+import net.ceedubs.ficus.Ficus.toFicusConfig
+import spray.json.JsObject
+
+import scala.concurrent.Await
+import scala.concurrent.duration.DurationInt
+import scala.util.{Failure, Success, Try}
+
+/**
+  * This class is set up to read a single (small!) DRS file into memory from our dev environment.
+  * Ensure that the Google account you're using has permission to access the DRS file in the
+  * environment you're targeting (use your test Google account in dev Terra!).
+  * (1) Log into your Google account locally: `gcloud auth login [your username]`
+  * (2) Log into Terra in your desired environment and make sure the repo you're using is linked
+  * (3) Ensure that the Martha instance below is the right one.
+  * (4) Set `drsPathToResolve` below to your DRS path
+  * (5) Update `mySecretFile` below to point to the local adc.json file for your Google account
+  * (6) "Run" NioChecksumTest using IntelliJ
+  * (7) Stop NioChecksumTest using IntelliJ because it does not stop on its own :(
+  */
+object NioChecksumTest {
+
+  /////////////////////
+  // UPDATE THESE
+  // V V V V V
+  private val marthaUrl = "https://us-central1-broad-dsde-dev.cloudfunctions.net/martha_v3"
+  private val drsPathToResolve = "drs://dg.4DFC:011a6a54-1bfe-4df9-ae24-990b12a812d3"
+  private val mySecretFile = "/Users/[USER]/.config/gcloud/legacy_credentials/[GOOGLE_ACCOUNT_NAME]/adc.json"
```

Comment:
TOL Not sure if it's worth polishing this up now but these could be environment variables or arguments to avoid recompilation.

Reply:
I was running this inside IntelliJ only, and the compilation time was small enough that I didn't notice it. For this very specific use case, I think it probably saves time to have them directly editable here rather than needing to deal with menus. Thanks for pointing this out, though, that's a pattern I'm not accustomed to and it's a good one to keep in mind for stuff like this.